File: app/src/main/java/com/vo1d/schedulemanager/v2/data/weeks/Week.java
package com.vo1d.schedulemanager.v2.data.weeks;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverters;

import com.vo1d.schedulemanager.v2.data.IMyEntity;
import com.vo1d.schedulemanager.v2.data.days.DaysOfWeek;

import java.util.List;

@Entity(tableName = "week_table", indices = {@Index(value = {"id"}, unique = true)})
public class Week implements IMyEntity, Comparable<Week> {

    @PrimaryKey(autoGenerate = true)
    public int id;

    public String title;

    @TypeConverters({DaysOfWeekToStringConverter.class})
    @ColumnInfo(defaultValue = "Monday,Tuesday,Wednesday,Thursday,Friday,Saturday")
    public List<DaysOfWeek> availableDays;

    public Week(String title) {
        this.title = title;
    }

    @Override
    public int compareTo(Week week) {
        return this.id - week.id;
    }
}

File: app/src/main/kotlin/ru/vo1d/ttmanager/data/entities/weeks/Week.kt
package ru.vo1d.ttmanager.data.entities.weeks

import androidx.room.Entity
import androidx.room.PrimaryKey
import ru.vo1d.ttmanager.data.DatabaseEntity
import ru.vo1d.ttmanager.data.DatabaseEntity.Companion.DEFAULT_ID

@Entity(tableName = "weeks")
data class Week(
    @field:PrimaryKey(autoGenerate = true)
    val id: Int = DEFAULT_ID,
    val title: String
) : DatabaseEntity
